services/order.py

```diff
@@ -0,0 +1,39 @@
+import datetime
+from typing import Optional
+
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket, User
+
+
+@transaction.atomic
+def create_order(
+        tickets: list,
+        username: str,
+        date: Optional[str] = None
+) -> Order:
+
+    user = User.objects.get(username=username)
+    if date is not None:
+        date = datetime.datetime.strptime(date, "%Y-%m-%d %H:%M")
+        order = Order.objects.create(user=user)
+        order.created_at = date
+        order.save()
+    else:
+        order = Order.objects.create(user=user)

```

Comment:
No need to duplicate code do as:
```suggestion
        date = datetime.datetime.strptime(date, "%Y-%m-%d %H:%M") if date is not None else None
        order = Order.objects.create(user=user, created_at=date)
```
Maybe like this?

Reply:
I fixed all, except last one, because an error raises.

Reply:
@ArsenPidhoretskyi  і @Bohdan-Salamakha , є надія, що сьогодні буду пити каву за завершення свого 9 модуля з цією задачею -:)

Reply:
@Zoriana-Dvulit It will be better if this is fixed. Let's talk in slack about this